### ISSUE TITLE
Provide health check API without certs required when TLS enabled

### ISSUE THREAD

**Issue by DanielZhangQD** [tikv/tikv#10297]
### Is your feature request related to a problem? Please describe:

When we deploy TiKV in the platforms like Kubernetes, we can leverage the health check mechanism provided by Kubernetes to exclude the unhealthy Pod from the service endpoint automatically, to do that, we need to provide a health check API.
We have the API /status, however, when TLS is enabled, certs are required to access this API, which is unavailable for Kubelet, so we need to provide a health check API without certs required when TLS enabled, e.g. /healthz.

**Comment by BusyJay** [comment]
This is also a health check grpc service in grpc that may help.

**Comment by DanielZhangQD** [comment]
The common probe handlers from Kubelet are TCP, HTTP, and HTTPS (even if it supports probing with a `command`, it still requires TLS certs and related tools to be installed).

**Comment by BusyJay** [comment]
Is it a common practice in k8s to provide such endpoint for gRPC based service?

**Comment by DanielZhangQD** [comment]
> Is it a common practice in k8s to provide such endpoint for gRPC based service?

@BusyJay Found the best practice https://github.com/grpc-ecosystem/grpc-health-probe/, we need to include that binary into the TiKV image.

**Comment by BusyJay** [comment]
I think this is a common requirement that may not be just useful for TiKV.

**Comment by DanielZhangQD** [comment]
OK, what's your recommendation? How should we do health check for TiKV in Kubernetes？

**Comment by BusyJay** [comment]
I agree we can use grpc-health-probe in Kubernetes. TiKV supports built-in grpc-health service since 5.0. What I argue is whether we should put it in TiKV image or a common tool image. For example, binaries like tikv-ctl, pd-ctl, br also need to be stored to some places.

**Comment by DanielZhangQD** [comment]
tikv-ctl is in the TiKV image, pd-ctl is in the PD image, BR/dumpling/tidb-lightning has its own image.
BTW, if the health check client is not in the TiKV image, Kubelet cannot use it to probe the status of TiKV process.
Another option is to install curl in the TiKV image and Kubelet calls curl to access the status API.

**Comment by BusyJay** [comment]
How do you check health state in PD or TiDB? Should grpc-health-probe be also included in those images?

**Comment by DanielZhangQD** [comment]
For TiDB, we use curl.
For PD, it has the similar issue now, use the grpc-health-probe or curl.
